Source/Particles/LaserParticleContainer.cpp

```diff
@@ -736,3 +742,67 @@ LaserParticleContainer::update_laser_particle (WarpXParIter& pti,
         }
         );
 }
+
+bool LaserParticleContainer::antenna_intersects_sim_box()
+{
+    const auto& warpx = WarpX::GetInstance();
+
+    const auto& geom = warpx.Geom(0);
+    const auto& sim_box = geom.ProbDomain();
+
+    std::vector<Real> t_corners;
+
+    const auto& x_l = sim_box.lo(0);
+    const auto& x_h = sim_box.hi(0);
+
+    const auto& x_p = m_position[0];
+    const auto& z_p = m_position[2];
+
+    const auto& x_n = m_nvec[0];
+    const auto& z_n = m_nvec[2];
+
+#if (defined WARPX_DIM_3D)
+    const auto& y_l = sim_box.lo(1);
+    const auto& y_h = sim_box.hi(1);
+    const auto& z_l = sim_box.lo(2);
+    const auto& z_h = sim_box.hi(2);
+
+    const auto& y_p = m_position[1];
+    const auto& y_n = m_nvec[0];
+
+    const auto ff = [&](auto x, auto y, auto z){
+        return (x-x_p)*x_n + (y-y_p)*y_n + (z-z_p)*z_n;
+    };
+
+    t_corners.push_back(ff(x_l,y_l,z_l));
+    t_corners.push_back(ff(x_l,y_l,z_h));
+    t_corners.push_back(ff(x_l,y_h,z_l));
+    t_corners.push_back(ff(x_l,y_h,z_h));
+    t_corners.push_back(ff(x_h,y_l,z_l));
+    t_corners.push_back(ff(x_h,y_l,z_h));
+    t_corners.push_back(ff(x_h,y_h,z_l));
+    t_corners.push_back(ff(x_h,y_h,z_h));
+
+#else
+    const auto& z_l = sim_box.lo(1);
+    const auto& z_h = sim_box.hi(1);
+
+    const auto ff = [&](auto x, auto z){
+        return (x-x_p)*x_n + (z-z_p)*z_n;
+    };
+
+    t_corners.push_back(ff(x_l, z_l));
+    t_corners.push_back(ff(x_l, z_h));
+    t_corners.push_back(ff(x_h, z_l));
+    t_corners.push_back(ff(x_h, z_h));
+#endif
```

Comment:
Awesome! Thanks a lot!

Do you want to add at least a check for the `z` direction for RZ, too?

Reply:
Yes! I was a bit lazy! But now I've added a simple check on `z` for the RZ case...